bumble/device.py

```diff
@@ -1172,8 +1172,17 @@ def get_services_by_uuid(self, uuid: core.UUID) -> List[gatt_client.ServiceProxy
         return self.gatt_client.get_services_by_uuid(uuid)
 
     def get_characteristics_by_uuid(
-        self, uuid: core.UUID, service: Optional[gatt_client.ServiceProxy] = None
+        self,
+        uuid: core.UUID,
+        service: Optional[gatt_client.ServiceProxy | core.UUID] = None,
```

Comment:
The '|' type syntax is only for Python >= 3.10, but we still support down to 3.8, so this should be using `Union`